src/main/java/com/ibm/g11n/pipeline/client/DocumentData.java

```diff
@@ -0,0 +1,117 @@
+/*  
+ * Copyright IBM Corp. 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ibm.g11n.pipeline.client;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * <code>DocumentData</code> provides read access to the document's
+ * properties.
+ * 
+ * @author John Emmons
+ */
+public abstract class DocumentData {
+    private final String documentId;
+    private final DocumentType type;
+    private final String sourceLanguage;
+    private final boolean readOnly;
+    private final String updatedBy;
+    private final Date updatedAt;
+
+    /**
+     * Protected constructor for a subclass extending <code>DocumentData</code>.
+     * 
+     * @param sourceLanguage    The source language of this bundle, specified by BCP 47
+     *                          language tag such as "en" for English.
+     */
+    public DocumentData(String documentId, DocumentType type, String sourceLanguage, boolean readOnly, String updatedBy, Date updatedAt) {
+        this.documentId = documentId;
+        this.type = type;
+        this.sourceLanguage = sourceLanguage;
+        this.readOnly = readOnly;
+        this.updatedBy = updatedBy;
+        this.updatedAt = updatedAt;
+    }
+    
+    /**
+     * Returns the source language of this document.
+     * 
+     * @return The source language of this document.
+     */
+    public final String getSourceLanguage() {
+        return sourceLanguage;
+    }
+
+    /**
+     * @return the readOnly
```

Comment:
Please provide API description. This comment applies to other places.

Reply:
ok done.